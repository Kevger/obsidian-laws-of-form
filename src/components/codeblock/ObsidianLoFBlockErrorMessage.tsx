import "./styles.css";

/**
 * Component to display error messages which appear in the code block
 *
 * @param msg  The error message to display
 * @returns A JSX element to display the error message
 */
export function ObsidianLoFBlockErrorMessage({
  msg,
}: {
  msg: string;
}): JSX.Element {
  return <div className="StyleObsidianLoFBlockErrorMessage">{msg}</div>;
}
