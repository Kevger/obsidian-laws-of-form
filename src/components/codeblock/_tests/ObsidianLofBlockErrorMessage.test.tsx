import { type ReactTestRenderer, create as render } from 'react-test-renderer';
import { ObsidianLoFBlockErrorMessage } from '../ObsidianLoFBlockErrorMessage';

describe('ObsidianLofBlockErrorMessage', () => {
  describe('given we have an error message', () => {
    const errorMessage = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
    let component: ReactTestRenderer;
    beforeEach(() => {
      component = render(<ObsidianLoFBlockErrorMessage msg={errorMessage} />);
    });

    it('renders correctly', () => {
      const snapshot = component.toJSON();
      expect(snapshot).toMatchSnapshot();
    });

    it('contains the error message', () => {
      expect(component.root.findByType('div').children[0]).toBe(errorMessage);
    });
  });
});
