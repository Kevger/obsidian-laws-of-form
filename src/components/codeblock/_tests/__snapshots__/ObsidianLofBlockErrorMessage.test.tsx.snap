// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObsidianLofBlockErrorMessage given we have an error message renders correctly 1`] = `
<div
  className="StyleObsidianLoFBlockErrorMessage"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</div>
`;
